### PR TITLE
webgpu: Optimize Softmax

### PR DESCRIPTION
**Pull request by qjia7**
Optimize softmax by fusing sub and exp.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7411)

### CODE REVIEW

**Review by mattsoulanille — APPROVED** [review]
LGTM

**Review by Linchenn — APPROVED** [review]
LGTM, thanks!
